src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -171,6 +171,15 @@ private function formatEditableCombinationsForListing(
                 $imageId = reset($defaultImageIds);
             }
 
+            if (null === $imageId) {
+                $imagePath = $this->productImagePathFactory->getNoImagePath(ProductImagePathFactory::IMAGE_TYPE_SMALL_DEFAULT);
+            } else {
+                $imagePath = $this->productImagePathFactory->getPathByType(
```

Comment:
@jolelievre is there any reason why this method doesn't handle getting a `NoImagePath` automatically?

Reply:
I'm not sure to understand what you mean? 🤔
You mean init `$imagePath` with fallback URL before the `if` ?

Reply:
Actually, it might not be that important. Right now you have to look everywhere if the `$imageId` is `null` and return the "No image path"`, but it's probably better to have it this way instead of checking `if null` inside `getPathByType` 👍

Reply:
Oh I get it, indeed the factory is stricter it expects a non nullable ImageId when available, and provide the `getNoImagePath` when you don't have one I think it's better to have two different identified behaviour to avoid confusion and internal magic
Then it's the responsibility of the "user code" to check its input data